tests/java/unit/com/dataiku/dss/formats/kml/KMLFormatTest.java

```diff
@@ -0,0 +1,198 @@
+
+import com.dataiku.dip.ApplicationConfigurator;
+import com.dataiku.dip.datalayer.Column;
+import com.dataiku.dip.datalayer.ColumnFactory;
+import com.dataiku.dip.datalayer.ProcessorOutput;
+import com.dataiku.dip.datalayer.Row;
+import com.dataiku.dip.datalayer.RowFactory;
+import com.dataiku.dip.datalayer.streamimpl.StreamColumnFactory;
+import com.dataiku.dip.datalayer.streamimpl.StreamRowFactory;
+import com.dataiku.dip.plugin.CustomFormatInput;
+import com.dataiku.dip.plugin.InputStreamWithContextInfo;
+
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class KMLFormatTest {
+
+    static class MockRow{
+        String name;
+        String description;
+        String geom;
+
+        public MockRow(String name, String description, String geom) {
+            this.name = name;
+            this.description = description;
+            this.geom = geom;
+        }
+
+        @Override
+        public String toString() {
+            return "MockRow{" +
+                    "name='" + name + '\'' +
+                    ", description='" + description + '\'' +
+                    ", geom='" + geom + '\'' +
+                    '}';
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            MockRow mockRow = (MockRow) o;
+            return Objects.equals(name, mockRow.name) &&
+                    Objects.equals(description, mockRow.description) &&
+                    Objects.equals(geom, mockRow.geom);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(name, description, geom);
+        }
+    }
+
+    static class RowAccumulator implements ProcessorOutput {
+
+        final Column colName;
+        final Column colDescription;
+        final Column colGeom;
+        final List<MockRow> rows = new ArrayList<>();
+
+        public RowAccumulator(ColumnFactory colFactory){
+            colName = colFactory.column("name");
+            colDescription = colFactory.column("description");
+            colGeom = colFactory.column("geom");
+        }
+
+        @Override
+        public void emitRow(Row row) throws Exception {
+            MockRow mockRow = new MockRow(row.get(colName), row.get(colDescription), row.get(colGeom));
+            rows.add(mockRow);
+        }
+
+        @Override
+        public void lastRowEmitted() throws Exception {
+
+        }
+
+        @Override
+        public void cancel() throws Exception {
+
+        }
+
+        @Override
+        public void setMaxMemoryUsed(long l) {
+
+        }
+    };
+
+    private CustomFormatInput kmlFormatInput;
+    private ColumnFactory colFactory = new StreamColumnFactory();
```

Comment:
As ColumnFactory and RowFactory maintains an internal state (the columns created and the rows), best is to init both member in setupDSS(), as we do for kmlFormatInput